docs/content/docs/build/data-modeling.md

```diff
@@ -3,19 +3,120 @@ title: Data modeling
+
+const newCell = await container.create(SharedCell); // Create a new SharedCell
+const newMap = await container.create(SharedMap); // Create a new SharedMap
+```
+
+### Using handles to store and retrieve Fluid objects
+
+All `LoadableObjects` have a `handle` property that can be used to store and retrieve them from other `LoadableObjects`. Objects created dynamically must be stored before they are collaborative. As you will see below, the act of storing a handle is what links the new dynamic object to the underlying data model and is how other clients learn that it exists.
+
+Dynamically created objects need to be stored on an already connected `LoadableObject`, so the most common case is to store them in an `initialObject`, because `initialObjects` are connected on creation. However, you can also store dynamic objects in other connected dynamic objects. In this sense `LoadableObjects` are arbitrarily nestable.
+
+When retrieving dynamically created objects you need to first get the object's handle then get the object from the handle. This reference based approach allows the Fluid Framework to virtualize the data underneath, only loading objects when they are requested.
+
+```typescript
+const schema = {
+    name: "example-container",
+    initialObjects: {
+        map: SharedMap,
+    },
+    dynamicObjectTypes: [ SharedCell ],
+}
+
+const { container, containerServices } = await client.getContainer(/*service config*/, schema);
+const map = container.initialObjects.map;
+
+const newCell = await container.create(SharedCell); // Create a new SharedCell
+map.set("cell-id", newCell.handle); // Attach the new SharedCell 
+
+// ...
+
+const cellHandle = map.get("cell-id"); // Get the handle
+const cell = await cellHandle.get(); // Resolve the handle to get the object
+
+// or
+
+const cell = await map.get("cell-id").get(); // Get and resolve handle
+
+// Listening for new dynamic objects
+
+map.on("valueChanged", (changed) => {
+    if (changed.key === "cell-id") {
+        const handle = map.get(changed.key);
+        handle.get().then((cell) => {
+            // ...
+        });
+    }
+}
+
+```
+
 ### When to use dynamic objects
+ 
+Dynamic objects are more difficult to work with than `initialObjects`, but are especially important for large data sets where portions of the data are virtualized. Because dynamic objects are loaded into memory on demand, using them can reduce boot time of your application by delaying when the objects are loaded.
+
+Dynamic objects are also not strictly defined in the container schema. This enables you to create containers with flexible, user-generated schemas.
 
-### How to create an object
+### Garbage collection on unreferenced objects
 
-### How to use handles to store objects
+Fluid automatically removes unreferenced objects from document snapshots. Cleaning up unreferenced data reduces the snapshot size, which leads to better download times and increases boot performance of a container.
```

Comment:
nit: let's not include snapshot and talk more about reduction of data size and how that helps boot.